### PR TITLE
Add initial development documentation

### PR DESCRIPTION
**Pull request by hasAdamr**

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Building
+
+Whenever templates are modified, the templates must be recompiled to binary, this is done with:
+
+```
+$ go generate github.com/TuneLab/gob/...
+```
+
+Then to build to your $GOPATH/bin dir, truss and it's two protoc plugins:
```
Comment (hasLeland) [inline DEVELOPING.md]: I think this line was intended to be:

> Then to build truss and it's two protoc plugins to your `$GOPATH/bin` directory:

**Comment by hasAdamr** [comment]
Fixed that line you mentioned @hasLeland

**Comment by hasLeland** [comment]
LGTM

File: DEVELOPING.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Building
+
+Whenever templates are modified, the templates must be recompiled to binary, this is done with:
+
+```
+$ go generate github.com/TuneLab/gob/...
+```
+
+Then to build truss and it's two protoc plugins to your $GOPATH/bin directory:
```
Comment (hasIan) [inline DEVELOPING.md:16]: _its_ not _it's_

**Pull request merged**